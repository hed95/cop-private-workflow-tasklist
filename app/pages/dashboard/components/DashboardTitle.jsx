import PropTypes from 'prop-types';
import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import * as actions from '../../../core/shift/actions';
import './DashboardTitle.css';
import AppConstants from '../../../common/AppConstants';
import secureLocalStorage from '../../../common/security/SecureLocalStorage';
import { endingShift, hasActiveShift } from '../../../core/shift/selectors';

class DashboardTitle extends React.Component {
  constructor(props) {
    super(props);
    this.endShift = this.endShift.bind(this);
    this.viewShift = this.viewShift.bind(this);
    this.secureLocalStorage = secureLocalStorage;
  }

  componentDidUpdate() {
    const { hasActiveShift } = this.props;
    if (!hasActiveShift) {
      window.location.reload();
    }
  }

  endShift(e) {
    e.preventDefault();
    this.secureLocalStorage.remove('shift');
    this.props.endShift();
  }

  viewShift(e) {
    e.preventDefault();
    const { history } = this.props;
    history.replace(AppConstants.SHIFT_PATH);
  }

  render() {
    const { endingShift, hasActiveShift, kc } = this.props;
    
    return (
      <div className="dashboard" id="dashboardTitle">
        <div className="govuk-width-container">
          <div className="govuk-grid-row dashboard__flex-container">
            <div className="dashboard__title">
              <h1 className="dashboard__text govuk-heading-xl">Welcome to the Central Operations Platform</h1>
            </div>

            {hasActiveShift && (
            <div className="dashboard__info-box">
              <h2 className="dashboard__text govuk-heading-m">My details</h2>
              <ul className="dashboard__text govuk-list">
                <li>
                  <span className="dashboard_text-heading">Name</span>
                  <span>{kc.tokenParsed.given_name} {kc.tokenParsed.family_name}</span>
                </li>
              </ul>
              <button
                id="editShift"
                className="govuk-button govuk-button--secondaryy dashboard__shift-button"
                type="submit"
                onClick={this.viewShift}
                disabled={endingShift}
              >Edit shift
              </button>
              <button
                id="endShift"
                className="govuk-button govuk-button--warning dashboard__shift-button"
                data-module="govuk-button"
                type="submit"
                onClick={this.endShift}
                disabled={endingShift}
                data-prevent-double-click="true"
              >End shift
              </button>
            </div>
          )}
            {!hasActiveShift && (
            <button id="startShift" className="govuk-button" type="submit" onClick={this.viewShift}>Start shift</button>
          )}
          </div>
        </div>
      </div>
    )
  }
}


DashboardTitle.propTypes = {
  endShift: PropTypes.func.isRequired,
  kc: PropTypes.shape({ logout: PropTypes.func }).isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

export default withRouter(connect(state => {
  return {
    kc: state.keycloak,
    hasActiveShift: hasActiveShift(state),
    endingShift: endingShift(state),
  };
}, mapDispatchToProps)(DashboardTitle));
