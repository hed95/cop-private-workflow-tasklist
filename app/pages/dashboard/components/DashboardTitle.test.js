import React from 'react';
import configureStore from 'redux-mock-store';
import Immutable from 'immutable';
import { DashboardTitle } from './DashboardTitle';

describe('DashboardTitle', () => {
  let props;

  beforeEach(() => {
    props = {
      isFetchingShift: false,
      hasActiveShift: true,
      shift: Immutable.fromJS({
        shiftid: 'shiftid',
      }),
      kc: {
        tokenParsed: {
          given_name: 'Givenname',
          family_name: 'Familyname'
        },
      },
      store: configureStore()({}),
    };
  });

  it('renders the title section correctly', async () => {
    const wrapper = await mount(<DashboardTitle {...props} />);

    const dashboardTitle = wrapper.find('#dashboardTitle');
    expect(dashboardTitle.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
