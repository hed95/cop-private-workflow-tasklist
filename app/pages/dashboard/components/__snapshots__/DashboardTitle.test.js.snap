// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTitle renders the title section correctly 1`] = `
<DashboardTitle
  hasActiveShift={true}
  isFetchingShift={false}
  kc={
    Object {
      "tokenParsed": Object {
        "family_name": "Familyname",
        "given_name": "Givenname",
      },
    }
  }
  shift={
    Immutable.Map {
      "shiftid": "shiftid",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="dashboard"
    id="dashboardTitle"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-grid-row dashboard__flex-container"
      >
        <div
          className="dashboard__title"
        >
          <h1
            className="dashboard__text govuk-heading-xl"
          >
            Welcome to the Central Operations Platform
          </h1>
        </div>
        <div
          className="dashboard__info-box"
        >
          <h2
            className="dashboard__text govuk-heading-m"
          >
            My details
          </h2>
          <ul
            className="dashboard__text govuk-list"
          >
            <li>
              <span
                className="dashboard_text-heading"
              >
                Name
              </span>
              <span>
                Givenname
                 
                Familyname
              </span>
            </li>
          </ul>
          <button
            className="govuk-button govuk-button--secondaryy dashboard__shift-button"
            id="editShift"
            onClick={[Function]}
            type="submit"
          >
            Edit shift
          </button>
          <button
            className="govuk-button govuk-button--warning dashboard__shift-button"
            data-module="govuk-button"
            data-prevent-double-click="true"
            id="endShift"
            onClick={[Function]}
            type="submit"
          >
            End shift
          </button>
        </div>
      </div>
    </div>
  </div>
</DashboardTitle>
`;
